src/services/device-management/iot-device.service.ts

```diff
@@ -1102,4 +1167,149 @@ export class IoTDeviceService {
             throw new BadRequestException(ErrorCodes.MissingABPInfo);
         }
     }
+
+    private async mapMQTTBrokerDevice(
+        iotDeviceDto: CreateIoTDeviceDto,
+        cast: MQTTBrokerDevice
+    ): Promise<MQTTBrokerDevice> {
+        const settings = iotDeviceDto.mqttBrokerSettings;
+        cast.authenticationType = settings.authenticationType;
+        switch (cast.authenticationType) {
+            case AuthenticationType.PASSWORD:
+                cast.mqttpassword ??= this.mqttService.hashPassword(
+                    settings.mqttpassword
+                );
+                cast.mqttusername = settings.mqttusername;
+                break;
+            case AuthenticationType.CERTIFICATE:
+                if (cast.deviceCertificate === undefined) {
+                    const certificateDetails = await this.mqttService.generateCertificate(
+                        cast.name
+                    );
+                    cast.deviceCertificate = certificateDetails.deviceCertificate;
+                    cast.deviceCertificateKey = this.encryptionHelperService.basicEncrypt(
+                        certificateDetails.deviceCertificateKey
+                    );
+                    cast.caCertificate = certificateDetails.ca;
+                    cast.mqttusername = iotDeviceDto.name;
```

Comment:
Shouldn't this be cast.name? The same that is send to generateCertificate and used as commonName :-)?

Reply:
It is the same value, but yes that will clarify that those two values has to be the same